regolith/helpers/u_milestonehelper.py

```diff
@@ -0,0 +1,138 @@
+    It can add a new milestone to the projecta collection.
+"""
+import datetime as dt
+from nameparser import HumanName
+import dateutil.parser as date_parser
+import sys
+import uuid
+
+from regolith.helpers.basehelper import DbHelperBase
+from regolith.fsclient import _id_key
+from regolith.tools import all_docs_from_collection
+
+
+TARGET_COLL = "projecta"
+
+def subparser(subpi):
+    subpi.add_argument("projectum_id", help="the id of the projectum")
+    subpi.add_argument("--number",
+                        help="number of the milestone to update from numbered list")
+    subpi.add_argument("--due_date",
+                       help="new due date of the milestone"
+                            "required to add a new milestone")
+    subpi.add_argument("-s", "--status",
+                       help="status of the milestone/deliverable:"
+                            "proposed, started, finished, back_burner"
+                            "paused, cancelled")
+    subpi.add_argument("--name",
+                       help="name of the new milestone."
+                            "required to add a new milestone.")
+    subpi.add_argument("-o", "--objective",
+                       help="objective of the new milestone."
+                            "required to add a new milestone")
+    subpi.add_argument("-t", "--type",
+                       help="type of the new milestone")
+    # Do not delete --database arg
+    subpi.add_argument("--database",
+                       help="The database that will be updated.  Defaults to "
+                            "first database in the regolithrc.json file.")
+    return subpi
+
+class MilestoneUpdaterHelper(DbHelperBase):
+    """Helper for updating milestones to the projecta collection.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "u_milestone"
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        rc = self.rc
+        key = rc.projectum_id
+
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) == 0:
+            raise RuntimeError(
+                "This entry appears to not exist in the collection")
+        filterid = {'_id': key}
+        current = rc.client.find_one(rc.database, rc.coll, filterid)
+        milestones = current.get('milestones', {})
+        deliverable = current.get('deliverable', {})
```

Comment:
I think you should add kickoff and deliverable to the list of milestones, they are not distinct.  Then sort the list by due-date, then just iterate over it.

You may then want to add a new field called "index" or something.  and assign it a value that is its place in the list.  Then when the program is rerun you can pull the milestone to be updated by comparing the value given by the user with the value of the index field.

Reply:
ok, I did that!